### ISSUE TITLE
Confusing Analyzer output when package language version is too old for null safety

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#42321]
In a package where the language version in `package_config.json` is < 2.9 and a library with a `// @dart = 2.9` comment, running `dartanalyzer --enable-experiment=non-nullable .` prints:

```
  error • This requires the 'non-nullable' language feature to be enabled. • foo.dart:10:20 • experiment_not_enabled
1 error found.
```

It would be less confusing if the error message said something about not being able to opt in to a newer language version with a comment than what is in the `package_config.json`. Or potentially even better, if it let you use the experiment anyway. This is on a recent dev SDK.

**Comment by natebosch** [comment]
See also #42323 where the CFE/VM have an output which is also confusing.

**Comment by scheglov** [comment]
@leafpetersen Is behavior of `// @dart =` specified anywhere?

Currently in analyzer we start with some set of language features, default, or with enabled experiments; and then this set passes through two phases of restricting - the first time based on the [package config](https://github.com/dart-lang/language/blob/master/accepted/future-releases/language-versioning/package-config-file-v2.md), and the second time based on the `// @dart` override.

So, if we specify a higher language version than for the whole package, it is no-op, the features have already been removed from the set.

It seems right from the perspective of migrating a package library by library.

**Comment by leafpetersen** [comment]
> @leafpetersen Is behavior of `// @dart =` specified anywhere?

[Here](https://github.com/dart-lang/language/tree/master/accepted/future-releases/language-versioning)

And more specifically [here](https://github.com/dart-lang/language/blob/master/accepted/future-releases/language-versioning/feature-specification.md#individual-library-language-version-override)

**Comment by scheglov** [comment]
Great, thank you!

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/151423

**Comment by scheglov** [comment]
One more related fix 60bb0d3a41b4004e29aba75005b30c5e8b6ae85c.
